### PR TITLE
feat core: refactor circuits_db and add some comments

### PR DESCRIPTION
**Pull request by vikria**

### CODE REVIEW

File: app/simplify.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ std::ifstream openInputFile(std::string const& file_path, csat::Logger& logger)
         std::cerr << "Can't open file, path is incorrect." << std::endl;
         std::abort();
     }
-    logger.error("File opened.");
+    logger.info("File opened.");
```
Comment (spefk) [inline app/simplify.cpp:82]: It should be `debug`

**Review by spefk — CHANGES_REQUESTED** [review]

File: src/simplification/utils/circuits_db.hpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                }
+
+                gates_operands[subcircuit_index].push_back(operands);
+            }
+            ++subcircuit_index;
+        }
+    }
+};
+
+struct DBSingleton
```
Comment (spefk) [inline src/simplification/utils/circuits_db.hpp:125]:
```c++
/**
 * Carries globally accecible pointers to the database of optimal circuits.
 */
```

File: src/simplification/utils/circuits_db.hpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+#include <string>
+#include <vector>
+
+#include "src/common/csat_types.hpp"
+#include "src/utility/converters.hpp"
+
+namespace csat::simplification
+{
+
+struct CircuitDB
```
Comment (spefk) [inline src/simplification/utils/circuits_db.hpp:20]: Needs docstring

**Review by spefk — CHANGES_REQUESTED** [review]

File: src/simplification/utils/circuits_db.hpp [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        read_db(db_path);
+    }
+
+    /**
+     * Reads a database in a BENCH format.
+     * Each row of the database must encode a circuit. Where:
+     *  -- the first number in the line is the number of inputs, which are numbered from 0 to this number - 1;
+     *  -- the second number is the number of outputs;
+     *  -- next "outputs" number of numbers are output codes, which are truth tables written in decimal form;
+     *  -- next "outputs" number of numbers are indexes of outputs;
```
Comment (Seva-Vaskin) [inline src/simplification/utils/circuits_db.hpp:53]: Not important note: "number of numbers" sounds wierd. 

Maybe: 
"The subsequent "outputs" entries represent output codes"

**Review by spefk — APPROVED** [review]

**Pull request merged**